lib/tests/it/main.rs

```diff
@@ -310,8 +326,26 @@ fn test_tar_export_structure() -> Result<()> {
     let src_tar = std::io::BufReader::new(std::fs::File::open(&src_tar)?);
     let mut src_tar = tar::Archive::new(src_tar);
     let expected = [
-        ("sysroot/ostree/repo/config", tar::EntryType::Regular, 0o644),
-        ("usr", tar::EntryType::Directory, libc::S_IFDIR | 0o755),
+        ("sysroot/ostree/repo", Directory, 0o755),
+        ("sysroot/ostree/repo/config", Regular, 0o644),
+        ("sysroot/ostree/repo/objects/00", Directory, 0o755),
+        ("sysroot/ostree/repo/objects/23", Directory, 0o755),
+        ("sysroot/ostree/repo/objects/77", Directory, 0o755),
+        ("sysroot/ostree/repo/objects/bc", Directory, 0o755),
+        ("sysroot/ostree/repo/objects/ff", Directory, 0o755),
+        ("sysroot/ostree/repo/refs", Directory, 0o755),
+        ("sysroot/ostree/repo/refs", Directory, 0o755),
+        ("sysroot/ostree/repo/refs/heads", Directory, 0o755),
+        ("sysroot/ostree/repo/refs/mirrors", Directory, 0o755),
+        ("sysroot/ostree/repo/refs/remotes", Directory, 0o755),
+        ("sysroot/ostree/repo/tmp", Directory, 0o755),
+        ("sysroot/ostree/repo/tmp/cache", Directory, 0o755),
+        (
+            "sysroot/ostree/repo/xattrs",
+            Directory,
+            libc::S_IFDIR | 0o755,
```

Comment:
This test fails, so I think we have a split-brain when we generate the tar content:
 * some directories like `sysroot/ostree/repo/xattrs` have mode `0o755`
 * some directories like `usr` have mode `libc::S_IFDIR | 0o755`

My current reading of `tar` specs is that the correct form is the first one, maybe?

Reply:
Right.  Today in ostree we end up mixing format and permissions in the mode bits.  I think this was a mistake.

Agree that our tar serialization should not inject mode bits into the permissions members, because tar already has a distinct field for that.

Reply:
Do you want to make this fix, or me?

Reply:
OK I pushed a commit to ignore it in tests

Reply:
Thanks for that. It was late evening already when I found out that the import path was also relying on modebits mixup.
I finished squashing the (hopefully) last buggy codepath today, PR at https://github.com/ostreedev/ostree-rs-ext/pull/220.